luigi/tools/range.py

```diff
@@ -217,101 +217,106 @@ def most_common(items):
-        regexes = [re.compile('(%04d).*(%02d).*(%02d).*(%02d)' % (d.year, d.month, d.day, d.hour)) for d in sample_datehours]
-        sample_tasks = [task_cls(d) for d in sample_datehours]
-        sample_outputs = [flatten_output(t) for t in sample_tasks]
-
-        for o, t in zip(sample_outputs, sample_tasks):
-            if len(o) != len(sample_outputs[0]):
-                raise NotImplementedError("Outputs must be consistent over time, sorry; was %r for %r and %r for %r" % (o, t, sample_outputs[0], sample_tasks[0]))
-                # TODO fall back on requiring last couple of days? to avoid astonishing blocking when changes like that are deployed
-                # erm, actually it's not hard to test entire hours_back..hours_forward and split into consistent subranges FIXME
-            for target in o:
-                if not isinstance(target, FileSystemTarget):
-                    raise NotImplementedError("Output targets must be instances of FileSystemTarget; was %r for %r" % (target, t))
-
-        for o in zip(*sample_outputs):  # transposed, so here we're iterating over logical outputs, not datehours
-            glob = cls._get_per_location_glob(sample_tasks, o, regexes)
-            yield o[0].fs, glob
-
-    @classmethod
-    def _list_existing(_, filesystem, glob, paths):
-        globs = _constrain_glob(glob, paths)
-        time_start = time.time()
-        listing = []
-        for g in sorted(globs):
-            logger.debug('Listing %s' % g)
-            listing.extend(filesystem.listdir(g))
-        logger.debug('%d %s listings took %f s to return %d items' % (len(globs), filesystem.__class__.__name__, time.time() - time_start, len(listing)))
-        return set(listing)
+    Convenient to use even from command line, like:
 
+        luigi --module your.module RangeHourly --of YourActualTask --start 2014-01-01T00
+    """
     def missing_datehours(self, task_cls, finite_datehours):
-        """Infers them by listing the task output target(s) filesystem.
-        """
-        filesystems_and_globs_by_location = self._get_filesystems_and_globs(task_cls)
-        paths_by_datehour = [[o.path for o in flatten_output(task_cls(d))] for d in finite_datehours]
-        listing = set()
-        for (f, g), p in zip(filesystems_and_globs_by_location, zip(*paths_by_datehour)):  # transposed, so here we're iterating over logical outputs, not datehours
-            listing |= self._list_existing(f, g, p)
-
-        # quickly learn everything that's missing
-        missing_datehours = []
-        for d, p in zip(finite_datehours, paths_by_datehour):
-            if not set(p) <= listing:
-                missing_datehours.append(d)
-
-        return missing_datehours
+        try:
+            return set(finite_datehours) - set(task_cls.bulk_complete.__func__(task_cls, finite_datehours))
```

Comment:
OK, let's fix that before merging. Is the issue that classmethods are not possible to overload? Could be something like that